### PR TITLE
Fold docs

### LINKED ISSUE #390: `Fold` docs

### PR DESCRIPTION
**Pull request by nomisRev**
closes #390

### CODE REVIEW

File: kategory-docs/docs/docs/optics/fold/README.md [hunk 1]
```diff
@@ -6,4 +6,58 @@ permalink: /docs/optics/fold/
 
 ## Fold
 
-TODO
\ No newline at end of file
+`Fold` is an optic that allows to focus into structure and get multiple results.
```
Comment (pakoito) [inline kategory-docs/docs/docs/optics/fold/README.md]: `Fold` is an optic that allows you to focus into and structure to get multiple results.

File: kategory-docs/docs/docs/optics/fold/README.md [hunk 2]
```diff
@@ -6,4 +6,58 @@ permalink: /docs/optics/fold/
...
+It is a generalisation of something `Foldable`.
```
Comment (pakoito) [inline kategory-docs/docs/docs/optics/fold/README.md]: It is a generalisation of the typeclass `Foldable`

File: kategory-docs/docs/docs/optics/fold/README.md [hunk 3]
```diff
@@ -6,4 +6,58 @@ permalink: /docs/optics/fold/
...
+val nestedNelFold: Fold<NonEmptyListKind<NonEmptyListKind<Int>>, NonEmptyListKind<Int>> = Fold.fromFoldable()
+
+val nestedNel = NonEmptyList.of(1, 2, 3, 4).map {
+    NonEmptyList.of(it, it)
+}
+
+(nestedNelFold compose nonEmptyIntFold).getAll(nestedNel)
+```
+
+`Fold` can be composed with all optics but `Setter` and result in the following optics.
```
Comment (pakoito) [inline kategory-docs/docs/docs/optics/fold/README.md]: `and results`, plural.

File: kategory-docs/docs/docs/optics/fold/README.md [hunk 4]
```diff
@@ -6,4 +6,58 @@ permalink: /docs/optics/fold/
 
 ## Fold
 
-TODO
\ No newline at end of file
+`Fold` is an optic that allows to focus into structure and get multiple results.
+It is a generalisation of something `Foldable`.
+
+Creating a `Fold` can be done by manually defining `foldMap`.
```
Comment (pakoito) [inline kategory-docs/docs/docs/optics/fold/README.md]: Could you please link back to Fold here?

File: kategory-docs/docs/docs/optics/fold/README.md [hunk 5]
```diff
@@ -6,4 +6,58 @@ permalink: /docs/optics/fold/
...
+```kotlin:ank
+nullableFold<Int>().headOption(null)
+```
+```kotlin:ank
+nonEmptyIntFold.headOption(NonEmptyList.of(1, 2, 3, 4))
+```
+
+## Composition
+
+Composing `Fold` can be used for accessing nested structures to be able to see foci.
```
Comment (pakoito) [inline kategory-docs/docs/docs/optics/fold/README.md]: `to be able to see foci` I don't quite understand what foci means

File: kategory-optics/src/main/kotlin/kategory/optics/Fold.kt [hunk 6]
```diff
@@ -17,9 +17,9 @@ import kategory.right
 import kategory.some
 
 /**
- * A [Fold] is an optic that allows to see into structure and get multiple results.
+ * A [Fold] is an optic that allows to focus into structure and get multiple results.
  *
- * [Fold] is a generalisation of [kategory.Foldable] and is implemented in terms of foldMap.
+ * [Fold] is a generalisation of something [Foldable] and is implemented in terms of foldMap.
```
Comment (pakoito) [inline kategory-optics/src/main/kotlin/kategory/optics/Fold.kt]: `an instance of Foldable` may fit better here.

File: kategory-docs/docs/docs/optics/fold/README.md (hunk 2 above)
Comment (pakoito) [reply kategory-docs/docs/docs/optics/fold/README.md]: Could you please link back to the Foldable documentation here?

File: kategory-docs/docs/docs/optics/fold/README.md (hunk 5 above)
Comment (nomisRev) [reply kategory-docs/docs/docs/optics/fold/README.md]: What I mean to say here is that if for example you have a Fold for `List<List<_>>` that can "focus" into `List<_>` and you compose it with a Fold for `List<Int>` with a focus in `Int` you get back a Fold with many focuses (foci). But now that I write it like this I am unsure if "foci" makes sense here.

Comment (pakoito) [reply kategory-docs/docs/docs/optics/fold/README.md]: `Composing `Fold` allows the user to create an aggregated foci (?) that focuses on nested structures`. Something in these terms maybe.

File: kategory-docs/docs/docs/optics/fold/README.md (hunk 1 above)
Comment (nomisRev) [reply kategory-docs/docs/docs/optics/fold/README.md]: I just checked the other docs and I am using "focus" in a different way then I did for other docs. To be consistent with other docs it should be:

A `Fold` is an optic that can see into a structure and get 0 to N foci. Not sure if that clears up the bit about nested structures. Thoughts?

Comment (pakoito) [reply kategory-docs/docs/docs/optics/fold/README.md]: Okay, consistency is better. Go for it!

**Pull request merged**